controller/api/public/stat_summary.go

```diff
@@ -398,14 +465,29 @@ func extractSampleValue(sample *model.Sample) uint64 {
 	return value
 }
 
-func metricToKey(metric model.Metric, groupBy model.LabelNames) string {
-	// this needs to match keys generated by MetaNamespaceKeyFunc
-	values := []string{}
+func metricToKey(metric model.Metric, groupBy model.LabelNames) pb.Resource {
+	// i.e. produce a key using the resource name and namespace
+	key := pb.Resource{}
+	resourceKeys := map[string]string{}
+	resourceName := ""
+
 	for _, k := range groupBy {
-		// return namespace/resource
-		values = append(values, string(metric[k]))
+		item := string(k)
+		if _, ok := k8s.ProxyLabelsToResourceTypes[item]; ok {
+			resourceKeys[item] = string(metric[k])
+			if item != "namespace" {
+				resourceName = item
+			}
+		}
 	}
-	return strings.Join(values, "/")
+
+	key.Name = resourceKeys[resourceName]
```

Comment:
I don't think this method is properly formulating keys for namespace-type resources. The stat integration test is failing on this branch with:

```
--- FAIL: TestCliStatForConduitNamespace (20.21s)
    --- PASS: TestCliStatForConduitNamespace/test_conduit_stat_deploy (0.09s)
    --- FAIL: TestCliStatForConduitNamespace/test_conduit_stat_namespace (20.12s)
    	stat_test.go:88: Expected success rate [100.00%] for [kltest], got [-]
```

The issue is that this method returns an empty key for the conduit namespace, whereas is should be returning a key with `{Name: <conduit-ns>}`.

I think we could fix and simplify this with something like:

```
func metricToKey(metric model.Metric, groupBy model.LabelNames) pb.Resource {
	// i.e. produce a key using the resource name and namespace
	key := pb.Resource{}

	for _, k := range groupBy {
		item := string(k)
		if _, ok := k8s.ProxyLabelsToResourceTypes[item]; ok {
			if item == "namespace" && len(groupBy) == 2 {
				key.Namespace = string(metric[k])
			} else {
				key.Name = string(metric[k])
			}
		}
	}

	return key
}
```

Reply:
Ah, and actually, if you make that change, then I don't think you need `ProxyLabelsToResourceTypes` at all, since `groupBy` will always contain either "namespace", or the string representing the name. In which case you could simplify even further to:

```
func metricToKey(metric model.Metric, groupBy model.LabelNames) pb.Resource {
	key := pb.Resource{}

	for _, k := range groupBy {
		if string(k) == "namespace" && len(groupBy) == 2 {
			key.Namespace = string(metric[k])
		} else {
			key.Name = string(metric[k])
		}
	}

	return key
}
```

That would also allow you to remove `ProxyLabelsToResourceTypes` and `reverseMap` from `pkg/k8s/k8s.go`.